src/mesh/Router.cpp

```diff
@@ -248,6 +248,13 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
     // If the packet is not yet encrypted, do so now
     if (p->which_payload_variant == meshtastic_MeshPacket_decoded_tag) {
         ChannelIndex chIndex = p->channel; // keep as a local because we are about to change it
+        auto hash = channels.setActiveByIndex(chIndex);
```

Comment:
The methods perhapsEncode() and perhapsDecode() are protected by a concurrency lockGuard, that prevents that the crypto lib (which is not reentrant) is run by two threads in parralel and possibly crashes. Now that you introduced encryption in another method (send) this possibly also requires a lockGuard for encryption here.

Reply:
good points. will use a different method here and not change perhapsEncode().

Reply:
I believe the `LockGuard` can be safely removed since we moved to ArduinoThread, because with that the device will never be simultaneously in two function calls. Seems to be a TODO since then: https://github.com/meshtastic/firmware/blob/4a867c81c05583a5f36259d60d38309eb8e936fe/src/concurrency/OSThread.h#L26

Reply:
> I believe the `LockGuard` can be safely removed since we moved to ArduinoThread, because with that the device will never be simultaneously in two function calls. Seems to be a TODO since then:
> 
> https://github.com/meshtastic/firmware/blob/4a867c81c05583a5f36259d60d38309eb8e936fe/src/concurrency/OSThread.h#L26

The lockguard was added later than that comment, i.e. fairly recently. The comment by @mverch67 about the BT Stacksize is valid too, we never really found out where the S3 crashes came from and suspected concurrency. 

My 2 cents: leave it there, it doesn't hurt and just eats a few cpu cycles.

Reply:
Ah, yes, it was added to the crypto function recently. Yes, we can keep it, but I don't think the change here will lead to crashes again.